woocommerce/woocommerce
# CYS: fix php warning raise by the Product Collection block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





These PR fixes ensure that the block editor state is updated when some blocks have to load some logic and set attributes dynamically in the editor when the assembler is loaded

Closes #42865.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new WooCommerce installation with this version.
2. Ensure that you have `WP_DEBUG` enabled.
3. Make sure to enable `customize-store` feature flag.
4. Go to `WooCommerce -> Home` and click `Customize Store` task.
5. Click `Start designing` button.
6. Follow through the flow all the way till the assembler hub shows up.
7. Go to FE.
8. Confirm there is no warning.

| Before | After |
|--------|--------|
| ![image](https://github.com/woocommerce/woocommerce/assets/4463174/23ea6733-90ab-4984-913e-a45f39bd5d39) | <img width="1219" alt="image" src="https://github.com/woocommerce/woocommerce/assets/4463174/f7a4df9e-b4c8-4325-a08d-4333080c83d6"> | 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS: fix php warning raise by the Product Collection block.

#### Comment 

</details>
